pip/pip-272.md

```diff
@@ -59,91 +32,41 @@ Pulsar Function supports three kinds of runtime and, correspondingly, has three
 
 # Motivation
 
-<!--
-Describe the problem this proposal is trying to solve.
-
-* Explain what is the problem you're trying to solve - current situation.
-* This section is the "Why" of your proposal.
--->
-
 Pulsar functions don't support configuring the `StateStoreProviderImplClass` with extra configurations, although there is a `Map<String, Object> config` parameter in the `StateStoreProvider#init` method, it's initialized as a map with only one field `stateStorageServiceUrl` in `javaInstanceRunnable`
 
 see:
 
-https://github.com/apache/pulsar/blob/master/pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java#L366-L369
+https://github.com/apache/pulsar/blob/0a39b819a9ec371322bfde61685497a2c21d9ab3/pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java#L366-L369
 
 Then when using a custom `StateStoreProvider`, it cannot get other configurations from the `init` method.
 
 # Goals
 
 ## In Scope
 
-<!--
-What this PIP intend to achieve once It's integrated into Pulsar.
-Why does it benefit Pulsar.
--->
-
 Make `StateStoreProvider` "truly" configurable, custom `StateStoreProvider` can be configured more easily and explicitly.
 
 ## Out of Scope
 
-<!--
-Describe what you have decided to keep out of scope, perhaps left for a different PIP/s.
--->
-
 
 # High Level Design
 
-<!--
-Describe the design of your solution in *high level*.
-Describe the solution end to end, from a birds-eye view.
-Don't go into implementation details in this section.
-
-I should be able to finish reading from beginning of the PIP to here (including) and understand the feature and 
-how you intend to solve it, end to end.
-
-DON'T
-* Avoid code snippets, unless it's essential to explain your intent.
--->
-
-Make `StateStoreProvider` configurable using custom configurations.
+We will add a configuration key to the Pulsar Function Worker configuration, containing the configuration for the `StateStoreProvider` implementation as a `Map<String, Object>`.
```

Comment:
I think people can't relate `map<String, object>` to the configuration of Pulsar Function Worker.  Map<string, object> is Java talk. As I replied in the comment, we need to fix this somehow.

Reply:
Maybe using `Map` here? As Pulsar functions use a YAML file for configuration, and YAML supports the `Map` struct

Reply:
Yes